envs/monkey_zoo/blackbox/analyzers/communication_analyzer.py

```diff
@@ -14,14 +15,22 @@ def __init__(self, island_client: MonkeyIslandClient, machine_ips: Iterable[str]
     def analyze_test_results(self):
         self.log.clear()
         all_monkeys_communicated = True
+        agent_ips = self._get_agent_ips()
+
         for machine_ip in self.machine_ips:
-            if not self.did_monkey_communicate_back(machine_ip):
-                self.log.add_entry("Monkey from {} didn't communicate back".format(machine_ip))
+            if not self._did_agent_communicate_back(machine_ip, agent_ips):
+                self.log.add_entry("Agent from {} didn't communicate back".format(machine_ip))
                 all_monkeys_communicated = False
             else:
-                self.log.add_entry("Monkey from {} communicated back".format(machine_ip))
+                self.log.add_entry("Agent from {} communicated back".format(machine_ip))
```

Comment:
We can rename and invert the logic to make it slightly more readable:

```
if self._agent_communicated_back():
    self.log.add_entry("Agent communicated back")
    return True

self.log.add_entry("Agent didn't communicate back")
return False
```

There are a bunch of variations on that, but the important thing I'm trying to get across is you can rename `_did_agent_communicate_back()` and invert the logic. If you do, your logic is almost an English sentence, "if agent communicated back, then...", as opposed to "If not did agent communicate back then"

Reply:
Not I would mind making that change 😉